extensions/spring-xd-extension-batch/src/main/java/org/springframework/batch/step/tasklet/x/AbstractProcessBuilderTasklet.java

```diff
@@ -224,21 +311,41 @@ protected String createClassPath(Class taskletClass) {
 		return lines;
 	}
 
-	protected void printLog(String commandName, List<String> lines, int exitCode) {
-		if (exitCode != 0) {
-			for (String line : lines) {
-				logger.error(commandName + " log: " + line);
+	protected void printLog(String commandName, List<String> out, List<String> err) {
```

Comment:
Going back to my previous comment about loading the files into memory, you could change this to take the `File` for out and err instead of reading it into a list first.